### PR TITLE
[PIL-780][OK] - Removed the default N/A for nonUKAddress PostalCode

### PR DESCRIPTION
**Pull request by olakunlekuye2**

### CODE REVIEW

File: app/uk/gov/hmrc/pillar2/models/hods/RegisterWithoutIDRequest.scala [hunk 1]
```diff
@@ -32,20 +32,33 @@ object NoIdOrganisation {
...
   implicit val addressFormat: OFormat[Address] = Json.format[Address]
-
   def fromAddress(address: UKAddress): Address =
-    Address(address.addressLine1, address.addressLine2, address.addressLine3, address.addressLine4, Some(address.postalCode), address.countryCode)
+    Address(
+      addressLine1 = address.addressLine1,
+      addressLine2 = address.addressLine2.filter(_.nonEmpty),
+      addressLine3 = address.addressLine3.filter(_.nonEmpty),
+      addressLine4 = address.addressLine4.filter(_.nonEmpty),
+      postalCode = address.postalCode.filter(_.nonEmpty),
```
Comment (TevynAllen) [inline app/uk/gov/hmrc/pillar2/models/hods/RegisterWithoutIDRequest.scala:49]: Don't see the need for the filter

File: app/uk/gov/hmrc/pillar2/models/hods/RegisterWithoutIDRequest.scala [hunk 2]
```diff
@@ -32,20 +32,33 @@ object NoIdOrganisation {
...
+    )
 
   def fromFmAddress(address: NonUKAddress): Address =
-    Address(address.addressLine1, address.addressLine2, address.addressLine3, address.addressLine4, address.postalCode, address.countryCode)
+    Address(
+      addressLine1 = address.addressLine1,
+      addressLine2 = address.addressLine2.filter(_.nonEmpty),
+      addressLine3 = address.addressLine3.filter(_.nonEmpty),
+      addressLine4 = address.addressLine4.filter(_.nonEmpty),
+      postalCode = address.postalCode.filter(_.nonEmpty),
```
Comment (TevynAllen) [inline app/uk/gov/hmrc/pillar2/models/hods/RegisterWithoutIDRequest.scala:59]: ditto

File: app/uk/gov/hmrc/pillar2/service/RegistrationService.scala [hunk 3]
```diff
@@ -101,7 +101,7 @@ class RegistrationService @Inject() (
       ultimateParentEntityName = registerWithoutIDRequest.organisation.organisationName,
       addressLine1 = registerWithoutIDRequest.address.addressLine1,
       addressLine2 = registerWithoutIDRequest.address.addressLine2.getOrElse(""),
-      townOrCity = registerWithoutIDRequest.address.addressLine3,
+      townOrCity = registerWithoutIDRequest.address.addressLine3.getOrElse(""),
```
Comment (alikouhestani) [inline app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:104]: @TevynAllen  I am more worried about this tbh

Comment (TevynAllen) [reply app/uk/gov/hmrc/pillar2/service/RegistrationService.scala:104]: Why? Its for the audit event

**Pull request closed without merging**